develop/2016-12-30-fgrehm-ocr-receipts-with-google-cloud-vision.py

```diff
@@ -0,0 +1,168 @@
+# 
+# For example, we can:
+# 
+# * Match the timestamp of a receipt in a city far away from Brasilia when we believe the congress person was supposed to be in a session.
+# * Match two timestamps of receipts made on the same day on cities really far from each other /
+# * Look for things like alcoholic beverages.
+# * See if the congressperson ordered too many dishes for "himself".
+# * Check in and check out dates from hotels.
+# 
+# Even though we have lots of libraries for doing OCR in Python, I believe [Google's Cloud Vision API](https://cloud.google.com/vision/) should be the "State of the art" when it comes to that type of thing since it is backed by Google, not to say that it is really easy to use. This notebook outlines the results of OCR'ing the receipts of the following 10 reimbursements picked from another analysis I did:
+# 
+# * https://jarbas.datasciencebr.com/#/document_id/5631309 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/1789/2015/5631309.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5631380 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/1789/2015/5631380.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5928875 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/1564/2016/5928875.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5768932 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/80/2015/5768932.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5962849 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/3052/2016/5962849.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5962903 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/3052/2016/5962903.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5855221 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/2238/2015/5855221.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5856784 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/2238/2015/5856784.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/5921187 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/2871/2016/5921187.pdf
+# * https://jarbas.datasciencebr.com/#/document_id/6069360 -> http://www.camara.gov.br/cota-parlamentar/documentos/publ/2935/2016/6069360.pdf
+# 
+# **NOTE**: While this could have been done in Python, it would take me a while to get it going so I kept things as simple as possible with bash since this is just a spike.
+
+# # Setup
+# 
+# Get an API key from Google Cloud that is able to access the [Vision API](https://cloud.google.com/vision/) and update the snippet below.
+# 
+# **MAKE SURE YOU DON'T COMMIT IT TO VERSION CONTROL**
+
+# In[1]:
+
+get_ipython().run_cell_magic('bash', '', '\nKEY="YOUR_KEY_HERE"\nKEY_PATH="/tmp/cloud-vision.key"\n\nif ! [[ -f $KEY_PATH ]]; then\n  echo "${KEY}" > $KEY_PATH\n  echo "Writing key to ${KEY_PATH}"\nelse\n  echo "Reusing existing key at ${KEY_PATH}"\nfi')
```

Comment:
Yup, its an alphanum with 39 characters, the only problem is that the notebook is written in bash. Will think about a way of making it work with minimal impact tomorrow.

From your end, you might need to enable cloud vision API for the project associated with the keys you're currently using for google maps. As a side note, I'm willing to share my keys with you and "donate" my US$300 free credit to the project in order to get all receipts OCRed if that's in the plans for the next 2 months before the credit expires 💵